### PR TITLE
Add priorityclasses for fleet in a list of resources to backup/restore

### PR DESCRIPTION
**Pull request by maxsokolovsky**
Issue: https://github.com/rancher/rancher/issues/40885

### CODE REVIEW

File: charts/rancher-backup/files/default-resourceset-contents/fleet.yaml
Comment (MbolotSuse) [inline charts/rancher-backup/files/default-resourceset-contents/fleet.yaml:24]: If my understanding is correct, priority classes are not namespaced, so this line does not help us here.

File: charts/rancher-backup/files/default-resourceset-contents/fleet.yaml
Comment (MbolotSuse) [inline charts/rancher-backup/files/default-resourceset-contents/fleet.yaml:22]: Right now this will get all priorityClasses - we only want to backup the one that the rancher chart specifies. You can do this with a `resourceNameRegexp` field, like [here](https://github.com/rancher/backup-restore-operator/blob/7e8dc2f00dad7879160be4f0bff2ec55b27fb215/charts/rancher-backup/files/default-resourceset-contents/fleet.yaml#L3). 

I would recommend anchoring the start and the end so that we exactly match what [rancher's chart](https://github.com/rancher/rancher/blob/a841a9e101ae601a033fa8b39342ad4800c91d85/chart/templates/priorityClass.yaml#L4) defines  - so something like `resourceNameRegexp: "^rancher-critical$"`

File: charts/rancher-backup/files/default-resourceset-contents/fleet.yaml
Comment (MbolotSuse) [inline charts/rancher-backup/files/default-resourceset-contents/fleet.yaml:1]: This priority class does not belong to fleet. Rather, it belongs to rancher's chart and is used by fleet. IMO, this more properly belongs in the [rancher](https://github.com/rancher/backup-restore-operator/blob/release/v3.0/charts/rancher-backup/files/default-resourceset-contents/rancher.yaml) resource set.

**Review by MbolotSuse — CHANGES_REQUESTED** [review]
A couple of comments on structure and implementation.

File: charts/rancher-backup/files/default-resourceset-contents/fleet.yaml
Comment (maxsokolovsky) [reply charts/rancher-backup/files/default-resourceset-contents/fleet.yaml:1]: Agreed!

**Review by MbolotSuse — APPROVED** [review]
One nit, but LGTM (up to you on if you want to fix the nit). Please wait for a review from team 3 before merging.

**Review by eliyamlevy — APPROVED** [review]

**Pull request merged**
